models/offer_template.go

```diff
@@ -16,7 +17,7 @@ import (
 //OfferTemplate contains the parameters of a template
 type OfferTemplate struct {
 	ID        string   `db:"id" json:"id" valid:"uuidv4"`
-	Key       string   `db:"key" json:"key" valid:"uuidv4,required"`
+	Key       string   `db:"key" json:"key" valid:"uuidv4"`
```

Comment:
Now I get key from the URL, so it is validated in params middleware and further added to the struct.